From 0b0cb6aa447cef0e4f631e43db05195d0d25329a Mon Sep 17 00:00:00 2001
From: Ryan Sundberg <ryan@arctype.co>
Date: Mon, 20 Jan 2025 15:17:39 -0800
Subject: [PATCH] crypto/tls: Disable test cases which are time variant

These test cases depend on the current date relative to the TLS
certificates under test. Skip them for repeatability.

--- FAIL: TestCrossVersionResume (0.02s)
    --- FAIL: TestCrossVersionResume/TLSv12 (0.01s)
        handshake_server_test.go:526: handshake did not resume at the same version
    --- FAIL: TestCrossVersionResume/TLSv13 (0.01s)
        handshake_server_test.go:526: handshake did not resume at the same version
FAIL
FAIL	crypto/tls	0.713s

How to patch:
cp to-root-project-dir
cat 0002-Skip-TLS-tests.patch | patch -p1
---
 src/crypto/tls/handshake_server_test.go | 4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/crypto/tls/handshake_server_test.go b/src/crypto/tls/handshake_server_test.go
index bbfe44bd97..4347c40a6d 100644
--- a/src/crypto/tls/handshake_server_test.go
+++ b/src/crypto/tls/handshake_server_test.go
@@ -493,8 +493,8 @@ func testSCTHandshake(t *testing.T, version uint16) {
 }

 func TestCrossVersionResume(t *testing.T) {
-	t.Run("TLSv12", func(t *testing.T) { testCrossVersionResume(t, VersionTLS12) })
-	t.Run("TLSv13", func(t *testing.T) { testCrossVersionResume(t, VersionTLS13) })
+	//t.Run("TLSv12", func(t *testing.T) { testCrossVersionResume(t, VersionTLS12) })
+	//t.Run("TLSv13", func(t *testing.T) { testCrossVersionResume(t, VersionTLS13) })
 }

 func testCrossVersionResume(t *testing.T, version uint16) {
--
2.41.0
